### PR TITLE
EPMDJ-7759 Handle errors when downloading a plugin

### PR DESCRIPTION
**Pull request by Mazel-Tovr**

### CODE REVIEW

File: core/src/test/kotlin/download/PluginDownloadTest.kt [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        val loaderService = ArtifactoryPluginLoader(Artifactory.GITHUB, pluginStorageDir, false)
+        loaderService.loadPlugins(listOf("unknown plugin"))
+        assert(pluginStorageDir.listFiles()?.size == 0)
+    }
+
+    @Test
+    fun `should successfully download only available plugins`() = runBlocking {
+        val loaderService = ArtifactoryPluginLoader(Artifactory.GITHUB, pluginStorageDir, false)
+        loaderService.loadPlugins(listOf("state-watcher", "unknown plugin"))
+        assert(pluginStorageDir.listFiles()?.size == 1)
```
Comment (LikhanovMaxim) [inline core/src/test/kotlin/download/PluginDownloadTest.kt]: Is it ok that tests spend a quota of downloading?

File: core/src/test/kotlin/download/PluginDownloadTest.kt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    @AfterTest
+    fun after() {
+        pluginStorageDir.deleteRecursively()
+    }
+
+    @Test
+    fun `should successfully download t2c plugin`() = runBlocking {
+        val loaderService = ArtifactoryPluginLoader(Artifactory.GITHUB, pluginStorageDir, false)
+        loaderService.loadPlugins(listOf("test2code"))
```
Comment (LikhanovMaxim) [inline core/src/test/kotlin/download/PluginDownloadTest.kt]: can you use allowedPlugins or another const?

File: core/src/test/kotlin/download/PluginDownloadTest.kt (hunk 1 above)
Comment (Mazel-Tovr) [reply core/src/test/kotlin/download/PluginDownloadTest.kt]: Well, in our case, when checks very often fall, this is not normal, I think it is worth turning off these tests until we fix the integration tests.

File: core/src/main/kotlin/plugins/ArtifactoryPluginLoader.kt [hunk 3]
```diff
@@ -27,14 +27,19 @@ private val logger = KotlinLogging.logger {}
...
         HttpClient(CIO).use { client ->
             for (pluginId in pluginIds) {
                 val url = pluginId.downloadUrl()
                 if (url.isNotBlank()) {
-                    client.downloadPlugin(url)
+                    runCatching {
+                        client.downloadPlugin(url)
+                    }.onFailure {
+                        logger.warn { "Specified url $url is incorrect, downloading from the specified repository ${artifactory.name}" }
+                        client.load(pluginId)
```
Comment (Mazel-Tovr) [inline core/src/main/kotlin/plugins/ArtifactoryPluginLoader.kt]: remove

**Review by LikhanovMaxim — CHANGES_REQUESTED** [review]
one runCatching

File: core/src/main/kotlin/plugins/PluginLoaderService.kt [hunk 4]
```diff
@@ -42,6 +44,7 @@ class PluginLoaderService(
     private val pluginPaths: List<File> = listOf(pluginStoragePath).map { it.canonicalFile }
 
     init {
+
```
Comment (ViktoriiaLebedeva) [inline core/src/main/kotlin/plugins/PluginLoaderService.kt]: remove it

**Review by ViktoriiaLebedeva — CHANGES_REQUESTED** [review]

**Review by LikhanovMaxim — APPROVED** [review]

**Pull request merged**